docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,264 @@
+# Swift and C++ Interoperability Roadmap
+
+## Introduction
+
+**This roadmap is** a sketch, rather than a final design for the “forward” (using C++ APIs from Swift) half of C++ and Swift interoperability. This roadmap outlines some high-level topics related to C++ interoperability, overarching goals that drive the project’s design decisions, a process for evolving C++ interoperability over time, and a collection of specific problems and their potential solutions. 
+
+“Reverse” interoperability (using Swift APIs from C++) is another extremely important part of the interoperability story, however, reverse interoperability has largely different goals and constraints, which necessarily mean a different design and roadmap. Alex Lorenz will be putting together a roadmap for reverse interoperability which outlines these goals and will shed light onto the differences between the two halves of the projects. In the future this roadmap will need to be updated to explain how these two halves fit together and present the “bi-directional” interoperability story that is C++ and Swift interoperability as a whole. 
+
+**C and Objective-C interoperability:** C++ interoperability will have to build off of the previous art of C and Objective-C interoperability, not only to preserve existing Swift codebases and provide a incremental story, but also to reduce unnecessary cognitive burden on developers. Starting with the name, C++ inherits much of C, which means C++ interoperability has a great starting place to build off of. C is a simple language without many high level constructs, so importing C APIs is, relatively, simple. On the other hand, Objective-C adds some high level constructs, such as reference types and a generics model. However, unlike C++, Objective-C has strong idioms; there are only so many patterns for doing some specific thing. So, while C and Objective-C interoperability is a good starting place, the model for importing these languages cannot be taken wholesale for C++ interoperability. 
```

Comment:
Maybe, add a line break after `**C and Objective-C interoperability:**`, so that  it acts more like an anchor / section header...

Reply:
@compnerd In the past you've mentioned the difference in the linkage model between C and C++ interop. Should this be mentioned out here?